src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -327,23 +347,32 @@ def _process_mash_tool(root_dir: str,
                 metadata = json.load(file)
 
             sketch_file = metadata['sketch_file']
+            sketch_data = _read_sketch(sketch_file)
+            sketches = sketch_data['sketches']
+            if len(sketches) != 1:
+                raise ValueError(f'Expected only one sketch in the mash info output for genome: {data_id}')
+            sketch_id = sketches[0]['name']
+            seq_meta.append({'sourceid': data_id, 'id': sketch_id})
```

Comment:
is there a reason  you chose to get the file name from the sketch rather than the metdata file? I'm not saying this is the wrong way to go, just curious.

I suppose if we really wanted to be safe we could check that that sketch and metadata file names are the same

Reply:
added the check. yea I was thinking reading from the sketch file is safe. I also don't know if the tool is always going to use file name as ID. But we added the check if it errors we will know.